File: base/src/main/java/com/windscribe/vpn/encoding/encoders/Encoder.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.vpn.encoding.encoders;

import java.io.IOException;
import java.io.OutputStream;

/**
 * Encode and decode byte arrays (typically from binary to 7-bit ASCII
 * encodings).
 */
public interface Encoder {

    int decode(String data, OutputStream out) throws IOException;

    int encode(byte[] data, int off, int length, OutputStream out) throws IOException;
}

File: base/src/main/java/com/windscribe/vpn/encoding/encoders/Encoder.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.vpn.encoding.encoders

import java.io.IOException
import java.io.OutputStream

/**
 * Encode and decode byte arrays (typically from binary to 7-bit ASCII
 * encodings).
 */
interface Encoder {
    @Throws(IOException::class)
    fun decode(
        data: String,
        out: OutputStream,
    ): Int

    @Throws(IOException::class)
    fun encode(
        data: ByteArray,
        off: Int,
        length: Int,
        out: OutputStream,
    ): Int
}
